frontend/src/diagram/palette/ContextualPalette.types.ts

```diff
@@ -10,14 +10,43 @@
  * Contributors:
  *     Obeo - initial API and implementation
  *******************************************************************************/
-import { GQLDeletionPolicy, GQLDiagramDescription, Tool, ToolSection } from 'diagram/DiagramWebSocketContainer.types';
+import { GQLDeletionPolicy, GQLToolSection, Tool } from 'diagram/DiagramWebSocketContainer.types';
 export interface ContextualPaletteProps {
-  diagramDescription: GQLDiagramDescription;
-  toolSections: ToolSection[];
-  targetElement: any;
+  editingContextId: string;
+  representationId: string;
+  diagramElement: any;
   invokeTool: (tool: Tool) => void;
   invokeConnectorTool: () => void;
   invokeLabelEdit: () => void | null;
   invokeDelete: (deletionPolicy: GQLDeletionPolicy) => void | null;
   invokeClose: () => void;
 }
+
+export interface GQLGetToolSectionsVariables {}
```

Comment:
It should include variables needed by the query. In that case, the definition should be:
```
export interface GQLGetToolSectionsVariables {
  editingContextId: string;
  diagramId: string;
  diagramElementId: string;
}
```

Reply:
Done